horace_core/sqw/@sqw/private/replicate_array.m

```diff
@@ -18,18 +18,16 @@
 
 % Original author: T.G.Perring
 
-if numel(npix) == numel(v)
-    % Get the bin index for each pixel
-    nend = cumsum(npix(:));
-    nbeg = nend - npix(:) + 1; % nbeg(i)=nend(i)+1 if npix(i)==0, but that's OK below
-    nbin = numel(npix);
-    npixtot = nend(end);
-    vout = zeros(npixtot, 1);
-    for i = 1:nbin
-        vout(nbeg(i):nend(i)) = v(i); % if npix(i)=0, this assignment does nothing
+if numel(npix)==numel(v)
+    if ~isempty(npix)
+        repl = arrayfun(@(x,y)ones(1,x)*y,npix(:),v(:),'UniformOutput',false);
+        vout = [repl{:}]';
+    else
+        vout=zeros(0,1);
     end
 else
-    error('SQW:replicate_array', ...
-        ['Number of elements in v must be equal to number of elements in ', ...
-        'npix. Found ''%i'' and ''%i'''], numel(v), numel(npix));
+    error('HERBERT:replicate_array:invalid_argument',...
```

Comment:
just checking - are we retaining HERBERT as an error indicator after the merge?

Reply:
probably left something from copying. I am not familiar with this function and do not recognize its meaning. Worth a simple ticket.  (and unit test)

Reply:
I've added #848 issue to reply to this question